elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java

```diff
@@ -206,35 +227,89 @@ private AtomicLoadKeyStore.LoadKey load(AtomicLoadKeyStore keyStore) throws Exce
         }
     }
 
-    private void checkCertificatesValidity(KeyStore keyStore) throws KeyStoreException {
+    private void checkCertificatesValidity(final KeyStore keyStore) throws KeyStoreException {
         if (ROOT_LOGGER.isEnabled(Logger.Level.WARN)) {
-            Enumeration<String> aliases = keyStore.aliases();
+            final Enumeration<String> aliases = keyStore.aliases();
             while (aliases.hasMoreElements()) {
-                String alias = aliases.nextElement();
-                Certificate certificate = keyStore.getCertificate(alias);
+                final String alias = aliases.nextElement();
+                final Certificate certificate = keyStore.getCertificate(alias);
                 if (certificate != null && certificate instanceof X509Certificate) {
-                    try {
-                        ((X509Certificate) certificate).checkValidity();
-                    } catch (CertificateExpiredException | CertificateNotYetValidException e) {
-                        ROOT_LOGGER.certificateNotValid(alias, e);
+                    final X509Certificate xCertificate = (X509Certificate) certificate;
+                    final CertificateValidity certificateValidity = CertificateValidity.getValidity(xCertificate.getNotBefore(), xCertificate.getNotAfter());
+                    switch(certificateValidity) {
+                        case ABOUT_TO_EXPIRE:
+                            ROOT_LOGGER.certificateAboutToExpire(keyStoreName, alias);
```

Comment:
We probably need a mechanism to avoid spamming the logs for the same certificate, especially if they want a frequent check.

Reply:
hmm, whats the point of check than ?